File: be-engine-v1/be-wallet/src/main/java/com/cmcorg20230301/be/engine/wallet/model/enums/SysUserWalletLogTypeEnum.java
package com.cmcorg20230301.be.engine.wallet.model.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.cmcorg20230301.be.engine.wallet.model.interfaces.ISysUserWalletLogType;
import com.fasterxml.jackson.annotation.JsonValue;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 用户钱包操作日志类型，枚举类
 */
@Getter
@AllArgsConstructor
public enum SysUserWalletLogTypeEnum implements ISysUserWalletLogType {

    ADD_PAY(101, "支付充值"), //

    ADD_BACKGROUND(102, "后台充值"), //

    ADD_TIME_CHECK(103, "超时返还"), //

    REDUCE_WITHDRAW(201, "用户提现"), //

    REDUCE_BACKGROUND(202, "后台扣除"), //

    REDUCE_USER_BUY(203, "用户购买"), // 这个购买的意思是：用户购买租户的钱包余额，租户扣除时的变化

    REDUCE_TENANT_BUY(204, "租户购买"), // 这个购买的意思是：租户购买租户的钱包余额，租户扣除时的变化

    ;

    @EnumValue
    @JsonValue
    private final int code; // 类型编码

    private final String name;

}

File: fe-antd-v1/src/model/enum/SysUserWalletLogTypeEnum.ts
import {IEnum} from "@/model/enum/CommonEnum";
import {ProSchemaValueEnumType} from "@ant-design/pro-components";

export interface ISysUserWalletLogTypeEnum {

    ADD_PAY: IEnum<number>,

    ADD_BACKGROUND: IEnum<number>,

    ADD_TIME_CHECK: IEnum<number>,

    REDUCE_WITHDRAW: IEnum<number>,

    REDUCE_BACKGROUND: IEnum<number>,

    REDUCE_USER_BUY: IEnum<number>,

    REDUCE_TENANT_BUY: IEnum<number>,

    ADD_USER_BUY_HASHRATE_PRODUCT_PROFIT_SHARE: IEnum<number>,

}

// 用户钱包操作日志类型，枚举类
export const SysUserWalletLogTypeEnum: ISysUserWalletLogTypeEnum = {

    ADD_PAY: {
        code: 101,
        name: '支付充值',
    },

    ADD_BACKGROUND: {
        code: 102,
        name: '后台充值',
    },

    ADD_TIME_CHECK: {
        code: 103,
        name: '超时返还',
    },

    REDUCE_WITHDRAW: {
        code: 201,
        name: '用户提现',
    },

    REDUCE_BACKGROUND: {
        code: 202,
        name: '后台扣除',
    },

    REDUCE_USER_BUY: {
        code: 203,
        name: '用户购买',
    },

    REDUCE_TENANT_BUY: {
        code: 204,
        name: '租户购买',
    },

    ADD_USER_BUY_HASHRATE_PRODUCT_PROFIT_SHARE: {
        code: 10001,
        name: '用户购买算力分成',
    },

}

export const SysUserWalletLogTypeEnumDict = new Map<number, ProSchemaValueEnumType>();

Object.keys(SysUserWalletLogTypeEnum).forEach(key => {

    const item = SysUserWalletLogTypeEnum[key];

    SysUserWalletLogTypeEnumDict.set(item.code as number, {text: item.name})

})
